src/Compilers/CSharp/Portable/Symbols/DelegateCaches/FullyConcreteChecker.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Checks whether a type or a method is fully constructed with fully concrete types as opposed to yet another generic parameter.
+    /// </summary>
+    internal class FullyConcreteChecker : CSharpSymbolVisitor<bool>
+    {
+        public static FullyConcreteChecker Instance { get; } = new FullyConcreteChecker();
+
+        public override bool VisitTypeParameter(TypeParameterSymbol symbol) => false;
+
+        public override bool VisitAlias(AliasSymbol symbol) => true;
+
+        public override bool VisitArrayType(ArrayTypeSymbol symbol) => Visit(symbol.ElementType);
+
+        public override bool VisitNamedType(NamedTypeSymbol symbol)
+        {
+            if (symbol.IsAnonymousType)
+            {
+                foreach (var anonPropType in AnonymousTypeManager.GetAnonymousTypePropertyTypes(symbol))
+                {
+                    if (!Visit(anonPropType))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            foreach (var typeArg in symbol.TypeArguments)
```

Comment:
Type signatures in metadata can specify optional ModOpt and required ModReq custom modifiers, which are types themselves. Sometimes they are inherited by declarations in source in  overriding/interface implementation cases. I don't think it is worth spending time creating a test case, but, if you really want to, you might get some ideas by looking at tests in src\compilers\csharp\test\symbol\symbols\custommodifierstests.cs. 
NamedTypeSymbol has two properties: one says if there are any custom modifiers for type arguments, the other aggregates them into an array of arrays. The names are, if I remember correctly, HasTypeArgumentsCustomModifiers and TypeArgumentsCustomModifiers. Look for examples how they are used in the codebase. 
I also forgot to mention that pointers and array can have custom modifiers too (see CustomModifiers property), they should be visited too.